reflex/event.py

```diff
@@ -704,24 +704,35 @@ def fn():
 
 def redirect(
     path: str | Var[str],
-    external: Optional[bool] = False,
+    external: Optional[bool] = None,
+    is_external: Optional[bool] = False,
     replace: Optional[bool] = False,
 ) -> EventSpec:
     """Redirect to a new path.
 
     Args:
         path: The path to redirect to.
         external: Whether to open in new tab or not.
+        is_external: Whether to open in new tab or not.
         replace: If True, the current page will not create a new history entry.
 
     Returns:
         An event to redirect to the path.
     """
+    if external is not None:
+        console.deprecate(
+            "The `external` prop in `rx.redirect`",
+            "use `is_external` instead.",
+            "0.6.6",
+            "0.7.0",
+        )
+        is_external = external
```

Comment:
hmm, if both `external` and `is_external` are specified, it seems maybe wrong if the deprecated arg takes over the value of the non-deprecated arg.

the code for sure gets uglier if both are `None` by default, but that way you'd be able to tell them apart. i'm on the fence about whether that's an improvement, but i wanted to mention it.

Reply:
How about doing something like this which ensures that `is_external` always takes precedence:

```python
def redirect(
    path: str | Var[str],
    external: Optional[bool] = None,
    is_external: Optional[bool] = None,
    replace: bool = False,
) -> EventSpec:
    """Redirect to a new path.

    Args:
        path: The path to redirect to.
        external: Whether to open in new tab or not.
        is_external: Whether to open in new tab or not.
        replace: If True, the current page will not create a new history entry.

    Returns:
        An event to redirect to the path.
    """

    if external is not None:
        console.deprecate(
            "The `external` prop in `rx.redirect`",
            "use `is_external` instead.",
            "0.6.6",
            "0.7.0",
        )

    if external is None and is_external is None:
        is_external = False

    if external is not None and is_external is None:
        is_external = external

    return server_side(
        "_redirect",
        get_fn_signature(redirect),
        path=path,
        external=is_external,
        replace=replace,
    )

```

Reply:
```py
is_external = (False if external is None else external) if is_external is None else is_external
```

Reply:
Not a huge fan of this one-liner but that works too

Reply:
yea i did it for fun more than practicality (i do like having the logic laid out without multiple branches though)